### PR TITLE
Add Data.Swagger

### PR DESCRIPTION
**Pull request by saralich**
Story details: https://app.clubhouse.io/itprotv/story/53932

### CODE REVIEW

**Review by vismaya-kalyan — APPROVED** [review]

File: Prolude/Swagger.hs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Prolude.Swagger 
+    ( Swagger.SwaggerType(Swagger.SwaggerString)
```
Comment (aschmois) [inline Prolude/Swagger.hs:6]: We should probably include all the types here even though we're not currently using them often.

File: Prolude/Swagger.hs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+overFirst f xs = case xs of
+  [] -> xs
+  x : ys -> f x : ys
+
+partialDropPrefix :: (Eq a, Show a) => [a] -> [a] -> [a]
+partialDropPrefix prefix list = case list of
+  [] -> []
+  _ -> Maybe.fromMaybe
+    (error (unwords [show prefix, "is not a prefix of", show list]))
+    (List.stripPrefix prefix list)
```
Comment (aschmois) [inline Prolude/Swagger.hs:79]: I think we can avoid these, they should be replaced by the new `Bauble` types.

File: Prolude/Swagger.hs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import qualified Data.String as String
+import qualified Data.Swagger as Swagger
+import qualified Data.Swagger.Declare as Swagger
+import qualified Data.Swagger.Internal.Schema as Swagger
+import qualified Data.Swagger.Internal.TypeShape as Swagger
+import qualified Data.Typeable as Typeable
+import qualified GHC.Generics as Generics
+
+type ToSwaggerParamSchema a = Swagger.ToParamSchema a
+type ToSwaggerSchema a = Swagger.ToSchema a
```
Comment (aschmois) [inline Prolude/Swagger.hs:28]: These definitely sound better but wouldn't be consistent with our other aliases, should we just name them consistently?

```suggestion
type SwaggerToParamSchema a = Swagger.ToParamSchema a
type SwaggerToSchema a = Swagger.ToSchema a
```

File: Prolude/Swagger.hs (hunk 2 above)
Comment (saralich) [reply Prolude/Swagger.hs:79]: After reviewing this I'm going to remove the addition of `withDefaultNamedSchema`, which will also remove the need for these helpers - it's only used twice within smurf, and if these functions are deprecated anyway, it seems like it'd be best not to add.

File: Prolude/Swagger.hs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    )
+where
+
+import qualified Data.Swagger as Swagger
+import qualified Data.Swagger.Declare as Swagger
+import qualified Data.Swagger.Internal.Schema as Swagger
+import qualified Data.Swagger.Internal.TypeShape as Swagger
+
+type SwaggerToParamSchema a = Swagger.ToParamSchema a
+type SwaggerToSchema a = Swagger.ToSchema a
```
Comment (tfausak) [inline Prolude/Swagger.hs:17]: Can these be eta reduced?
```suggestion
type SwaggerToParamSchema = Swagger.ToParamSchema
type SwaggerToSchema = Swagger.ToSchema
```

File: prolude.cabal [hunk 5]
```diff
@@ -47,12 +48,14 @@ library
...
         , generic-random >= 1.3.0.1 && < 1.4
         , mongoDB >= 2.7.0 && < 2.8
+        , mtl >= 2.2.2 && < 2.3
         , QuickCheck >= 2.14.1 && < 2.15
         , quickcheck-instances >= 0.3.25 && < 0.4
         , persistent >= 2.8 && < 3
         , persistent-mongoDB >= 2.8 && < 3
         , safe-exceptions >= 0.1.7 && < 0.2
         , scientific >= 0.3.6 && < 0.4
+        , swagger2 >= 2.6 && < 3
```
Comment (tfausak) [inline prolude.cabal:58]:
```suggestion
        , swagger2 >= 2.6 && < 2.7
```

**Review by tfausak — APPROVED** [review]

**Review by tfausak — APPROVED** [review]

**Pull request merged**